InternalServerListPing event throwing a null exception

EssentialsX version (`/essentials`): 2.0.1-b514

Server software (`/version`): Paper 1172 (1.12.1)

Since I updated both Paper and EssentialsX to support 1.12.1, I got this stacktrace spammed in my console on every server I own :
```
[StatusProcessingThread/ERROR]: Could not pass event InternalServerListPingEvent to Essentials v2.0.1-b514
org.bukkit.event.EventException: null
        at com.massivecraft.massivelag.inject.MassiveLagEventExecutor.execute(MassiveLagEventExecutor.java:79) ~[?:?]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[patched_1.12.1.jar:git-Paper-1172]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:514) ~[patched_1.12.1.jar:git-Paper-1172]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:496) ~[patched_1.12.1.jar:git-Paper-1172]
        at protocolsupport.protocol.packet.handler.AbstractStatusListener.lambda$handleStatusRequest$1(AbstractStatusListener.java:68) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_131]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_131]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_131]
Caused by: java.lang.NullPointerException
        at com.earth2me.essentials.EssentialsServerListener.onServerListPing(EssentialsServerListener.java:43) ~[?:?]
        at sun.reflect.GeneratedMethodAccessor64.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_131]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_131]
        at com.massivecraft.massivelag.inject.MassiveLagEventExecutor.execute(MassiveLagEventExecutor.java:62) ~[?:?]
        ... 7 more
```

Everything seems to work though, but my log size is growing very fast.

--- a/Essentials/src/com/earth2me/essentials/EssentialsServerListener.java
+++ b/Essentials/src/com/earth2me/essentials/EssentialsServerListener.java
@@ -58,8 +58,10 @@ public class EssentialsServerListener implements Listener {
             } catch (NullPointerException e) {
                 if (!npeWarned) {
                     npeWarned = true;
-                    throw new Exception("A plugin has fired a ServerListPingEvent "
+                    Exception ex = new Exception("A plugin has fired a ServerListPingEvent "
                             + "without implementing Paper's methods. Point the author to https://git.io/v7Xzl.");
+                    ex.setStackTrace(e.getStackTrace());
+                    throw ex;
                 }
             }
         } else {

--- a/Essentials/src/com/earth2me/essentials/EssentialsServerListener.java
+++ b/Essentials/src/com/earth2me/essentials/EssentialsServerListener.java
@@ -17,7 +17,8 @@ import java.util.logging.Level;
 
 public class EssentialsServerListener implements Listener {
     private final transient IEssentials ess;
-    private boolean errorLogged = false;
+    private boolean unsupportedLogged = false;
+    private boolean npeWarned = false;
     private boolean isPaperSample;
     private Method setSampleText;
     private Method getSampleText;
@@ -36,7 +37,7 @@ public class EssentialsServerListener implements Listener {
     }
 
     @EventHandler(priority = EventPriority.LOWEST)
-    public void onServerListPing(final ServerListPingEvent event) {
+    public void onServerListPing(final ServerListPingEvent event) throws Exception {
         if (isPaperSample) {
             try {
                 List<String> playerNames = (List<String>) getSampleText.invoke(event, null);
@@ -49,9 +50,16 @@ public class EssentialsServerListener implements Listener {
                 }
                 setSampleText.invoke(event, playerNames);
             } catch (IllegalAccessException | InvocationTargetException | ClassCastException e) {
-                if (!errorLogged) {
-                    ess.getLogger().log(Level.WARNING, "Unable to hide players from server list ping using Paper 1.12+ method!", e);
-                    errorLogged = true;
+                if (!unsupportedLogged) {
+                    ess.getLogger().log(Level.WARNING, "Unable to hide players from server list ping "
+                            + "using Paper 1.12+ method!", e);
+                    unsupportedLogged = true;
+                }
+            } catch (NullPointerException e) {
+                if (!npeWarned) {
+                    npeWarned = true;
+                    throw new Exception("A plugin has fired a ServerListPingEvent "
+                            + "without implementing Paper's methods. Point the author to https://git.io/v7Xzl.");
                 }
             }
         } else {
@@ -64,10 +72,10 @@ public class EssentialsServerListener implements Listener {
                     }
                 }
             } catch (UnsupportedOperationException e) {
-                if (!errorLogged) {
+                if (!unsupportedLogged) {
                     ess.getLogger().warning("Current server implementation does not support "
                             + "hiding players from server list ping. Update or contact the maintainers.");
-                    errorLogged = true;
+                    unsupportedLogged = true;
                 }
             }
         }

